dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/dataitems/DataItemQueryTests.java

```diff
@@ -0,0 +1,297 @@
+    }
+
+    @Test
+    public void testGetAllDataItemsUsingDefaultPaginationOrderedByCode()
+    {
+        // When
+        final ApiResponse response = dataItemActions.get( "?order=code:asc" );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects[0].dimensionItemType", isA( String.class ) );
+        response.validate().body( "nameableObjects[0].code", is( "AAAAAAA-1234" ) );
+    }
+
+    @Test
+    public void testFilterByDimensionTypeUsingDefaultPagination()
+    {
+        // Given
+        final String theDimensionType = "PROGRAM_INDICATOR";
+        final String theUrlParams = "?filter=dimensionItemType:in:[%s]";
+
+        // When
+        final ApiResponse response = dataItemActions.get( format( theUrlParams, theDimensionType ) );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects[0].dimensionItemType", is( theDimensionType ) );
+    }
+
+    @Test
+    public void testFilterByProgramUsingDefaultPagination()
+    {
+        // Given
+        final String theDimensionType = "PROGRAM_INDICATOR";
+        final String theProgramId = "BJ42SUrAvHo";
+        final String aValidFilteringAttribute = "program.id";
+        final String theUrlParams = "?filter=dimensionItemType:in:[%s]&filter=" + aValidFilteringAttribute
+            + ":eq:%s&order=code:asc";
+
+        // When
+        final ApiResponse response = dataItemActions.get( format( theUrlParams, theDimensionType, theProgramId ) );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects[0].code", is( "AAAAAAA-1234" ) );
+    }
+
+    @Test
+    public void testFilterUsingInvalidDimensionTypeUsingDefaultPagination()
+    {
+        // Given
+        final String anyInvalidDimensionType = "INVALID_TYPE";
+        final String theUrlParams = "?filter=dimensionItemType:in:[%s]";
+
+        // When
+        final ApiResponse response = dataItemActions.get( format( theUrlParams, anyInvalidDimensionType ) );
+
+        // Then
+        response.validate().statusCode( is( CONFLICT ) );
+        response.validate().body( "pager", is( nullValue() ) );
+        response.validate().body( "httpStatus", is( "Conflict" ) );
+        response.validate().body( "httpStatusCode", is( CONFLICT ) );
+        response.validate().body( "status", is( "ERROR" ) );
+        response.validate().body( "errorCode", is( "E2016" ) );
+        response.validate().body( "message", containsString(
+            "Unable to parse element `INVALID_TYPE` on filter `dimensionItemType`. The values available are:" ) );
```

Comment:
you could use anyInvalidDimensionType var here instead of INVALID_TYPE

Reply:
Done